src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs

```diff
@@ -11,29 +15,29 @@ public static IServiceCollection IdentityApplicationBuilderExtensions(this IServ
             services
                 .AddAuthorization(options =>
                 {
-                    options.AddPolicy("admin", policy => policy.RequireClaim("Admin"));
+                    options.AddPolicy(UserPolicy.User, policy => policy.RequireAssertion(context => context.User.IsInRole("User")));
+                    options.AddPolicy(UserPolicy.Admin, policy => policy.RequireAssertion(context => context.User.IsInRole("Admin")));
                 })
                 .AddAuthentication(options => {
                     options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
                     options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
                 })
                 .AddJwtBearer(options =>
                 {
-                    options.Authority = "https://localhost:5123";
+                    options.Authority = "https://localhost:5001";
+                    options.Audience = "resourceOwner";
                     options.TokenValidationParameters = new TokenValidationParameters
                     {
                         //ValidateIssuerSigningKey = true,
-                        //ValidateAudience = true,
-                        //ValidateIssuer = true,
-                        RoleClaimType = "role",
-                        NameClaimType = "sub",
+                        ValidateAudience = false,
+                        ValidateIssuer = true,
+                        RoleClaimType = ClaimTypes.Role,

```

Comment:
Aspnet core default ClaimType.Role value (`http://schemas.microsoft.com/ws/2008/06/identity/claims/role`). is very different, then we are passing in JWT token. This is why Authorize attribute wasn't working for you. Use  
```
RoleClaimType = "role",
NameClaimType = "sub",
```

Reply:
at the start, I was using the way that you described but it was not working that's why I have swapped to ClaimType.Role below is the image of the HttpContext.User.Claims which is coming with ClaimType.Role
![image](https://user-images.githubusercontent.com/57753967/110327883-feb69c80-8033-11eb-9795-ecf167c106bc.png)

Reply:
with `RoleClaimType = ClaimTypes.Role` the `[Authorize(Roles = "User")]` works properly

Reply:
Ok, I will double check. For now this is fine.